File: CalWatchAndroid/app/src/main/java/com/calwatch/android/api/models/Report.java
package com.calwatch.android.api.models;

import java.util.ArrayList;

/**
 * Created by John R. Kosinski on 30/1/2559.
 */
public class Report {
    private ArrayList<ReportDay> days = new ArrayList<>();
    private int totalCalories;
    private int averageCaloriesPerDay;
    private int numberOfDaysOverTarget;
    private FilterParams filterParams;

    public int getTotalCalories() {
        return totalCalories;
    }
    public void setTotalCalories(int totalCalories) {
        this.totalCalories = totalCalories;
    }

    public int getAverageCaloriesPerDay() {
        return averageCaloriesPerDay;
    }
    public void setAverageCaloriesPerDay(int averageCaloriesPerDay) {
        this.averageCaloriesPerDay = averageCaloriesPerDay;
    }

    public int getNumberOfDaysOverTarget() {
        return numberOfDaysOverTarget;
    }
    public void setNumberOfDaysOverTarget(int numberOfDaysOverTarget) {
        this.numberOfDaysOverTarget = numberOfDaysOverTarget;
    }

    public ArrayList<ReportDay> getDays() {
        return days;
    }
    public void setDays(ArrayList<ReportDay> days) {
        this.days = days;
    }

    public FilterParams getFilterParams(){
        return filterParams;
    }
    public void setFilterParams(FilterParams filterParams){
        this.filterParams = filterParams;
    }
}

File: CalWatchApi/CalWatch.Api/Models/Report.cs
using System;
using System.Collections.Generic;

namespace CalWatch.Api.Models
{
    public class Report
    {
        private List<ReportDay> days = new List<ReportDay>(); 

        /// <summary>
        /// Total calories for entire report period.
        /// </summary>
        public long TotalCalories { get; set; }
        /// <summary>
        /// Average per day, in the time period requested.
        /// </summary>
        public long AverageCaloriesPerDay { get; set; }
        public int NumberOfDaysOverTarget { get; set; }
        /// <summary>
        /// List of days involved in compiling the report.
        /// </summary>
        public List<ReportDay> Days
        {
            get
            {
                return days;
            }
        }
        public FilterParams FilterParams { get; set; }
    }
}
